File: app/src/main/java/org/baiyu/fuckshare/filetype/VideoType.java
package org.baiyu.fuckshare.filetype;

import java.util.Map;
import java.util.Set;

public enum VideoType implements FileType {
    MP4("mp4", Set.of(
            Map.of(0, new byte[]{(byte) 0x66, (byte) 0x74, (byte) 0x79, (byte) 0x70, (byte) 0x69, (byte) 0x73, (byte) 0x6F, (byte) 0x6D})
    )),
    AVI("avi", Set.of(
            Map.of(0, new byte[]{(byte) 0x52, (byte) 0x49, (byte) 0x46, (byte) 0x46},
                    8, new byte[]{(byte) 0x41, (byte) 0x56, (byte) 0x49, (byte) 0x20}
            )
    )),
    MPEG("mpeg", Set.of(
            Map.of(0, new byte[]{(byte) 0x00, (byte) 0x00, (byte) 0x01, (byte) 0xB3})
    ));

    private final String extension;
    private final Set<Map<Integer, byte[]>> signatures;


    VideoType(String extension, Set<Map<Integer, byte[]>> signatures) {
        this.extension = extension;
        this.signatures = signatures;
    }

    @Override
    public String getExtension() {
        return extension;
    }

    @Override
    public Set<Map<Integer, byte[]>> getSignatures() {
        return signatures;
    }
}

File: app/src/main/java/org/baiyu/fuckshare/filetype/VideoType.kt
package org.baiyu.fuckshare.filetype

@Suppress("SpellCheckingInspection")
enum class VideoType(
    override val extension: String,
    override val signatures: Set<Map<Int, ByteArray>>
) : FileType {
    MP4(
        "mp4", setOf(
            mapOf(
                4 to
                        "ftypisom".toByteArray(Charsets.US_ASCII)
            ),
            mapOf(
                4 to
                        "ftypiso2".toByteArray(Charsets.US_ASCII)
            ),
            mapOf(
                4 to
                        "ftypiso5".toByteArray(Charsets.US_ASCII)
            ),
            mapOf(
                4 to
                        "ftypmmp4".toByteArray(Charsets.US_ASCII)
            ),
            mapOf(
                4 to
                        "ftypmp41".toByteArray(Charsets.US_ASCII)
            ),
            mapOf(
                4 to
                        "ftypmp42".toByteArray(Charsets.US_ASCII)
            ),
            mapOf(
                4 to
                        "ftypavc1".toByteArray(Charsets.US_ASCII)
            )
        )
    ),
    AVI(
        "avi", setOf(
            mapOf(
                0 to "RIFF".toByteArray(Charsets.US_ASCII),
                8 to "AVI ".toByteArray(Charsets.US_ASCII)
            )
        )
    ),
    MPEG(
        "mpeg", setOf(
            mapOf(
                0 to
                        byteArrayOf(0x00.toByte(), 0x00.toByte(), 0x01.toByte(), 0xB3.toByte())
            ),
            mapOf(
                0 to
                        byteArrayOf(0x00.toByte(), 0x00.toByte(), 0x01.toByte(), 0xBA.toByte())
            )
        )
    ),
    MKV(
        "mkv", setOf(
            mapOf(
                0 to
                        byteArrayOf(0x1A.toByte(), 0x45.toByte(), 0xDF.toByte(), 0xA3.toByte())
            )
        )
    );
}